mirascope/chat/models.py

```diff
@@ -90,3 +90,83 @@ def stream(
 
         for chunk in completion_stream:
             yield OpenAIChatCompletionChunk(chunk=chunk)
+
+
+class AsyncOpenAIChat:
```

Comment:
I would refactor this into the following structure:

```
chat/models/
    __init__.py
    async_openai_chat.py
    openai_chat.py
    utils.py
```

Make sure to then update the `__init__` everywhere so imports still work as desired.

Then I would consider pulling the `if tools:` block into a utility function. If you don't think that's worthwhile, don't include the `utils.py` addition.

Reply:
i'll hold off on adding utils for now